### ISSUE TITLE
Autominimier / Dist Files

### ISSUE THREAD

**Issue by codeclinic** [davidsneighbour/colossus-wp-options-framework#97]
Do we have any means by which js etc can be autominimised? This would save us having to commit both and release both dev files and there minimised versions.

**Comment by davidsneighbour** [comment]
The repo has a gulpfile.js, but it's neither working with the original package.json nor with my changes. I would create a new build system with npm scripts, that works better for me in other cases (themes and custom plugins).

My plan is to have a build process that takes this all out of our hands. optimum (in my opinion) is one single file per field type that is loaded dynamically only when that field type is required. I did not get to the place in the code where the JavaScripts are loaded, this is maybe already included in some form. 

Then maybe a wp-config.php constant like CF_ISDEV that loads the unminimized files on request.

What do you think?

**Comment by codeclinic** [comment]
Personally I don't see the need to distribute the uncompressed files. Most people are using this because they don't want the hassle of digging into the code side of making options. Also, 99.9% of enduser are the site owners using a plugin or theme that in-turn uses Titan/Colossus. They don't need or want the uncompressed files, and in most cases don't even want it. Therefore the smaller the plugin the better, and the harder it is for none-devs to mess with code, the better.

Another benefit of only shipping minimised files is that we can use "smallness" as a usp to dev looking for a framework to use with their plugin/theme.

My preference would be to have Github releases which just contained the absolute minimum kb for devs to incorporate into their themes/plugins. This would then be reflected in the Colossus plugin.

**Comment by davidsneighbour** [comment]
I agree 100%. The releases and packages (plugin/theme) will have only minified files. But the idea I am having requires a way to have those per-field files unminified in the repo. So for developers, I would expect them to clone the repo and work with the build script to get an unminified version so they can debug more properly (I never get these map files working in my projects somehow).

The main idea, if I did not make that clear enough, is that I think it might be a good idea to include the scripts/libraries for dropdowns only if the theme/plugin adds a dropdown. so each field that requires additional scripts will have those in a single file that is loaded when the field is initiated (speed, size). But maybe that's overkill. From the looks of it there are plenty of javascript libraries in the inc folder that will blow up the size. The more minified in a release, the better.

**Comment by davidsneighbour** [comment]
todo: add npm script to minify javascript and stylesheets in their current setup/combination and exclude unminified files from published packages.
